### PR TITLE
Move to beginning / end

### PR DESCRIPTION
**Pull request by kettch**
Adding support for moveToBeginningOfDocument / moveToEndOfDocument

### CODE REVIEW

File: JNWCollectionView/JNWCollectionViewFramework.m [hunk 1]
```diff
@@ -429,6 +429,30 @@ - (NSArray *)allIndexPaths {
 	return indexPaths.copy;
 }
 
+- (NSIndexPath *)firstIndexPath {
+	NSIndexPath *indexPath = nil;
```
Comment (jwilling) [inline JNWCollectionView/JNWCollectionViewFramework.m]: I wonder if this method can be simplified to something like this?

``` objc
if ([[self numberOfItemsInSection:0] > 0) {
	return [NSIndexPath jnw_indexPathForItem:0 inSection:0];
}

return nil;
```

File: JNWCollectionView/JNWCollectionViewFramework.m [hunk 2]
```diff
@@ -1093,6 +1118,20 @@ - (void)moveLeftAndModifySelection:(id)sender {
 	[self selectItemAtIndexPath:toSelect atScrollPosition:JNWCollectionViewScrollPositionNearest animated:YES selectionType:JNWCollectionViewSelectionTypeExtending];
 }
 
+- (void)moveToBeginningOfDocument:(id)sender {
+	NSIndexPath *toSelect = [self firstIndexPath];
+	if (toSelect) {
```
Comment (jwilling) [inline JNWCollectionView/JNWCollectionViewFramework.m]: There is already a nil check in `selectItemAtIndexPath:atScrollPosition:animated`.

File: JNWCollectionView/JNWCollectionViewFramework.m [hunk 3]
```diff
@@ -1093,6 +1118,20 @@ - (void)moveLeftAndModifySelection:(id)sender {
...
+		[self selectItemAtIndexPath:toSelect atScrollPosition:JNWCollectionViewScrollPositionNearest animated:YES];
+	}
+}
+
+- (void)moveToEndOfDocument:(id)sender {
+	NSIndexPath *toSelect = [self lastIndexPath];
+	if (toSelect) {
```
Comment (jwilling) [inline JNWCollectionView/JNWCollectionViewFramework.m]: Same note as above.

**Comment by jwilling** [comment]
Thanks! 🍨  Just a couple comments.

File: JNWCollectionView/JNWCollectionViewFramework.m (hunk 1 above)
Comment (kettch) [inline JNWCollectionView/JNWCollectionViewFramework.m]: True, updating it.

File: JNWCollectionView/JNWCollectionViewFramework.m (hunk 3 above)
Comment (kettch) [inline JNWCollectionView/JNWCollectionViewFramework.m]: Removed both.

**Comment by jwilling** [comment]
Awesome! Thanks. 👍

**Pull request merged**